cfme/cloud/flavor.py

```diff
@@ -108,11 +142,59 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=True)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            self.wait_for_disappear()
+
+    def wait_for_disappear(self):
+        """Wait for flavor to disappear"""
+        wait_for(self.provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=300)
+        wait_for(lambda: not self.exists,
+                 timeout=300,
+                 message='Wait for flavor to disappear',
+                 delay=20,
+                 fail_func=self.browser.refresh)
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except FlavorNotFound:
+            return False
+
 
 @attr.s
 class FlavorCollection(BaseCollection):
     ENTITY = Flavor
 
+    def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True):
+        view = navigate_to(self, 'Add')
+        form_params = {'provider': provider.name,
+                       'flavor_name': name,
+                       'ram_size': ram,
+                       'vcpus': vcpus,
+                       'disk_size': disk,
+                       'swap_size': swap,
+                       'rxtx_factor': rxtx,
+                       'public': is_public}
+
+        view.fill(form_params)
+        view.add.click()
+        view.flash.assert_success_message(
```

Comment:
When you add a new flavor, the page is redirected to the parent view. I believe you would have to create a new view here correct?

Reply:
I intended to get flavor object to be returned. Does instantiate creates a new one ?

Reply:
Also i saw examples where instantiate is used:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/storage/volume.py#L333